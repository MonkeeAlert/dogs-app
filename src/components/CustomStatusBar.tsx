import {
  View,
  StyleSheet,
  ColorValue,
  SafeAreaView,
  StatusBar,
  StatusBarProps,
  Platform,
} from 'react-native';
import React from 'react';
import {useSafeAreaInsets} from 'react-native-safe-area-context';

interface Props extends StatusBarProps {
  bg: ColorValue;
}

const STATUSBAR_HEIGHT = StatusBar.currentHeight;
const APPBAR_HEIGHT = Platform.OS === 'ios' ? 44 : 56;

const CustomStatusBar = ({bg: backgroundColor, ...props}: Props) => {
  const insets = useSafeAreaInsets();

  return (
    <View style={[{height: insets.top}, {backgroundColor}]}>
      <StatusBar translucent bg={backgroundColor} {...props} />
    </View>
  );
};

const styles = StyleSheet.create({
  statusBar: {
    height: STATUSBAR_HEIGHT,
  },
  appBar: {
    backgroundColor: 'red',
    height: APPBAR_HEIGHT,
  },
  content: {
    flex: 1,
    backgroundColor: '#33373B',
  },
});

export default CustomStatusBar;
