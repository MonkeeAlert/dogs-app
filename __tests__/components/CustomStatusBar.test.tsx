import React from 'react';
import renderer from 'react-test-renderer';
import CustomStatusBar from '../../src/components/CustomStatusBar';

describe('CustomStatusBar', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<CustomStatusBar backgroundColor={'transparent'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
