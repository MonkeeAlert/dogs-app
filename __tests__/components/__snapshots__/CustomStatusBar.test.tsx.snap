// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStatusBar matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "height": 44,
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
  testID="StatusBar"
/>
`;
